apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammarBuilder.java

```diff
@@ -0,0 +1,284 @@
+     */
+    public ApexGrammarBuilder rule(GrammarRuleKey ruleKey) {
+        validateRule(ruleKey);
+        currentRule = ruleKey;
+        return this;
+    }
+
+    /**
+     * Creates and stores and parsing expression "single".
+     *
+     * @param object expression.
+     * @return an ApexGrammarBuilder instance.
+     */
+    public ApexGrammarBuilder is(Object object) {
+        return addExpression(convertToExpression(object));
+    }
+
+    /**
+     * Creates and stores and parsing expression "sequence".
+     *
+     * @param object first expression.
+     * @param rest rest of expressions.
+     * @return an ApexGrammarBuilder instance.
+     */
+    public ApexGrammarBuilder is(Object object, Object... rest) {
+        return addExpression(new SequenceExpression(convertToExpressions(Lists.asList(object, rest))));
+    }
+
+    /**
+     * Creates parsing expression "first of".
+     *
+     * @param first first expression.
+     * @param second second expression.
+     * @return an Expression.
+     */
+    public ParsingExpression firstOf(Object first, Object second) {
+        return new FirstOfExpression(convertToExpression(first), convertToExpression(second));
+    }
+
+    /**
+     * Creates parsing expression "first of" by a sequence.
+     *
+     * @param first first expression.
+     * @param second second expression.
+     * @param rest rest of expressions.
+     * @return an Expression
+     */
+    public ParsingExpression firstOf(Object first, Object second, Object... rest) {
+        return new FirstOfExpression(convertToExpressions(Lists.asList(first, second, rest)));
+    }
+
+    /**
+     * Creates parsing expression "optional".
+     *
+     * @param object expression.
+     * @return an Expression.
+     */
+    public ParsingExpression optional(Object object) {
+        return new OptionalExpression(convertToExpression(object));
+    }
+
+    /**
+     * Builds and returns a grammar instance, built on a specific grammar builder.
+     *
+     * @return a Grammar
+     */
+    public Grammar build() {
+        Grammar result;
+        if (isFulGrammar) {
+            LexerfulGrammarBuilder grammarBuilder = LexerfulGrammarBuilder.create();
+            mapRules.forEach((rule, expression) -> {
+                grammarBuilder.rule(rule).is(expression);
+            });
+            grammarBuilder.setRootRule(rootRule);
+            result = grammarBuilder.build();
+        } else {
+            LexerlessGrammarBuilder grammarBuilder = LexerlessGrammarBuilder.create();
+            mapRules.forEach((rule, expression) -> {
+                grammarBuilder.rule(rule).is(expression);
+            });
+            grammarBuilder.setRootRule(rootRule);
+            result = grammarBuilder.build();
+        }
+        return result;
+    }
+
+    /**
+     * Gets a map rules. Only available for unit test.
+     *
+     * @return a map rules.
+     */
+    Map<GrammarRuleKey, ParsingExpression> getMapRules() {
```

Comment:
maybe you can rename it to "getMapRules_forTesting"
this way we can know if it's valud for use it or not